src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs

```diff
@@ -68,6 +68,11 @@ public override PeerAddress FindPeerToConnectTo()
                 }
 
                 peer = this.peerAddressManager.PeerSelector.SelectPeer();
+                if (peer == null)
+                {
+                    peerSelectionFailed++;
+                    continue;
+                }
 
                 if (!peer.NetworkAddress.Endpoint.Address.IsValid())
```

Comment:
why not 
`if ((peer == null) || !peer.NetworkAddress.Endpoint.Address.IsValid())`

Reply:
OK so I wanted to leave the conditions separate so that I can put logging in each "fail" scenario.

Let me add this now.